backend/app/core/circuit_breaker.py

```
@@ -0,0 +1,232 @@
+
+    def decorator(func: Callable) -> Callable:
+        @wraps(func)
+        async def wrapper(*args, **kwargs):
+            try:
+                # Use pybreaker's built-in async support
+                return await breaker.call_async(func, *args, **kwargs)
+            except pybreaker.CircuitBreakerError as e:
+                CIRCUIT_BREAKER_REJECTED.labels(breaker_name=breaker.name).inc()
+                logger.error(
+                    f"[CircuitBreaker] {breaker.name} is OPEN, rejecting request"
+                )
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)